File: spring-cloud-integrated-demo/order-service/src/main/java/com/github/jccode/springcloud/integrateddemo/order/controller/OrderController.java
package com.github.jccode.springcloud.integrateddemo.order.controller;

import com.github.jccode.springbootsample.core.data.rest.RestResult;
import com.github.jccode.springcloud.integrateddemo.order.form.OrderForm;
import com.github.jccode.springcloud.integrateddemo.order.model.Order;
import com.github.jccode.springcloud.integrateddemo.order.service.OrderService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.validation.BindingResult;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

import static com.github.jccode.springbootsample.core.utils.CommonUtil.success;
import static com.github.jccode.springbootsample.core.utils.WebUtil.checkResult;

@RestController
public class OrderController {

    @Autowired
    private OrderService orderService;


    /**
     * 下单;
     *
     * curl -XPOST -d "userId=1&address=LianhuaRoad&items[0].productId=3&items[0].amount=100"
     *
     * @param orderForm
     * @return
     */
    @PostMapping("/reserve")
    public RestResult reserveOrder(@Valid OrderForm orderForm, BindingResult result) {
        checkResult(result);
        Order order = orderService.reserveOrder(orderForm);
        return success(order);
    }

    // 支付确认.
    @PostMapping("/confirm")
    public RestResult confirmOrder() {
        return null;
    }
}

File: spring-cloud-integrated-demo/order-service/src/main/scala/com/github/jccode/springcloud/integrateddemo/order/controller/OrderController.scala
package com.github.jccode.springcloud.integrateddemo.order.controller

import javax.validation.Valid

import com.github.jccode.springbootsample.core.data.RestResult
import com.github.jccode.springbootsample.core.data.RestResult.RestSuccess
import com.github.jccode.springbootsample.core.utils.WebUtil
import com.github.jccode.springcloud.integrateddemo.order.form.OrderForm
import com.github.jccode.springcloud.integrateddemo.order.model.Order
import com.github.jccode.springcloud.integrateddemo.order.service.OrderService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.{PostMapping, RestController}

@RestController
class OrderController(@Autowired val orderService: OrderService) {

  @PostMapping(Array("/reserve"))
  def reserveOrder(@Valid orderForm: OrderForm, result: BindingResult): RestSuccess[Order] = {
    WebUtil.checkResult(result)
    RestResult.success(orderService.reserveOrder(orderForm))
  }

  @PostMapping(Array("/confirm"))
  def confirmOrder() = ???

}
